docs/proposals/14-wireguard-based-cluster-vpn.md

```diff
@@ -0,0 +1,178 @@
+
+# Cluster VPN based on Wireguard with reversed connection
+
+# Problem Statement
+
+Today's Gardener cluster VPN solution has several major issues including:
+
+1. Connection establishment is always from the seed cluster to the shoot cluster. This means that there needs to be connectivity both ways which is not desirable in many cases (OpenStack, VMware) and causes high effort in firewall configuration or extra infrastructure. These firewall configurations are prohibited in some cases due to security policies.
+2. Shoot clusters must provide a VPN endpoint. This means extra cost for the endpoint (roughly €20/month on hyperscalers) or will consume scarce resources (limited number of VMware NSX-T load balancers).
+
+Issue 1 above currently prevents us from rolling out our infrastructure efficiently on OpenStack.
+
+A first implementation has been provided to resolve the issues with the [konnectivity server](https://github.com/gardener/gardener/blob/master/docs/proposals/11-apiserver-network-proxy.md). As we did find several severe shortcomings with the underlying technology component, the [apiserver-network-proxy](https://github.com/kubernetes-sigs/apiserver-network-proxy) we believe that this is not a suitable way ahead. We have opened an [issue](https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/180) and provided two solution proposals to the community but judging from the responsiveness the component appears not to be maintained regularly. We do see some remedies, e.g. using the [Quick Protocol](https://de.wikipedia.org/wiki/Quick_UDP_Internet_Connections) instead of GRPC but we (a) consider the implementation effort significantly higher compared to this proposal and (b) would use an experimental protocol to solve a problem that can also be solved with existing and proven core network technologies.
```

Comment:
I understand that in this proposal you decided against following up with Konnectivity/the apiserver-network-proxy. What will happen to today's implementation? Will you remove it entirely? If we are not planning to use Konnectivity then it makes little sense to keep all this code (which already complicates e.g. the KAPI deployment).

Reply:
Yes I would remove it but only if we jointly agree on a way forward.

Reply:
Of course. Can you then add these plan outlines to the document so that it's clear for everybody?